From 9413c5a46cb30af432e54d5e7f908f302fd5112e Mon Sep 17 00:00:00 2001
From: Laur59 <148864407+Laur59@users.noreply.github.com>
Date: Fri, 14 Feb 2025 15:48:23 +0200
Subject: [PATCH 2/2] Suppress architecture AArch64 from the toolchain

(cherry picked from commit 8bc69dfeafec52942bfb2782cdcea844d4cd595d)

build_scripts/build-from-arm-toolchain.sh -w 3
clang version 20.1.0 (upstream 9f24451bad0) 2025-02-14
Target: arm-unknown-none-eabi
Thread model: posix
InstalledDir: /Users/scratch/todelete/embedded/cross/llvm-current/arm-newlib/bin
Built from :
  ATfE upstream     9f24451bad0
  ATfE branch       release-arm-software20-for-ukos
build_scripts/build-from-arm-toolchain.sh -w 3

(cherry picked from commit e72faddf6a58b598a8a1d6465650b0f85d7808d0)
---
 arm-software/embedded/CMakeLists.txt          |   4 +-
 .../embedded/arm-multilib/json/multilib.json  | 120 ------------------
 .../arm-multilib/multilib-generate.py         |   7 +-
 3 files changed, 3 insertions(+), 128 deletions(-)

diff --git a/arm-software/embedded/CMakeLists.txt b/arm-software/embedded/CMakeLists.txt
index 48d266bb530e..ad2ef59e1b03 100644
--- a/arm-software/embedded/CMakeLists.txt
+++ b/arm-software/embedded/CMakeLists.txt
@@ -235,8 +235,8 @@ set(LLVM_TOOLCHAIN_DISTRIBUTION_COMPONENTS
 installed by the install-llvm-toolchain target"
 )
 set(LLVM_ENABLE_PROJECTS clang;lld CACHE STRING "")
-set(LLVM_TARGETS_TO_BUILD AArch64;ARM CACHE STRING "")
-set(LLVM_DEFAULT_TARGET_TRIPLE aarch64-linux-gnu CACHE STRING "")
+set(LLVM_TARGETS_TO_BUILD ARM CACHE STRING "")
+set(LLVM_DEFAULT_TARGET_TRIPLE arm-none-eabi CACHE STRING "")
 set(LLVM_APPEND_VC_REV OFF CACHE BOOL "")
 set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "")
 set(LLVM_ENABLE_ZLIB OFF CACHE BOOL "")
diff --git a/arm-software/embedded/arm-multilib/json/multilib.json b/arm-software/embedded/arm-multilib/json/multilib.json
index 20d860c97f0e..726b53260617 100644
--- a/arm-software/embedded/arm-multilib/json/multilib.json
+++ b/arm-software/embedded/arm-multilib/json/multilib.json
@@ -1,125 +1,5 @@
 {
     "libs": [
-        {
-            "variant": "aarch64a_exn_rtti_unaligned",
-            "json": "aarch64a_exn_rtti_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf"
-        },
-        {
-            "variant": "aarch64a_unaligned",
-            "json": "aarch64a_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf -fno-exceptions -fno-rtti"
-        },
-        { 
-            "variant": "aarch64a_be_exn_rtti",
-            "json": "aarch64a_be_exn_rtti.json",
-            "flags": "--target=aarch64_be-unknown-none-elf"
-        },
-        {
-            "variant": "aarch64a_be",
-            "json": "aarch64a_be.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -fno-exceptions -fno-rtti"
-        },
-        {
-            "variant": "aarch64a_exn_rtti",
-            "json": "aarch64a_exn_rtti.json",
-            "flags": "--target=aarch64-unknown-none-elf -mno-unaligned-access"
-        },
-        {
-            "variant": "aarch64a",
-            "json": "aarch64a.json",
-            "flags": "--target=aarch64-unknown-none-elf -mno-unaligned-access -fno-exceptions -fno-rtti"
-        },
-        {
-            "variant": "aarch64a_soft_nofp_exn_rtti",
-            "json": "aarch64a_soft_nofp_exn_rtti.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64a_soft_nofp",
-            "json": "aarch64a_soft_nofp.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft -fno-exceptions -fno-rtti",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64a_be_soft_nofp_exn_rtti",
-            "json": "aarch64a_be_soft_nofp_exn_rtti.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64a_be_soft_nofp",
-            "json": "aarch64a_be_soft_nofp.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft -fno-exceptions -fno-rtti",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_exn_rtti_unaligned",
-            "json": "aarch64r_exn_rtti_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r"
-        },
-        {
-            "variant": "aarch64r_unaligned",
-            "json": "aarch64r_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -fno-exceptions -fno-rtti"
-        },
-        {
-            "variant": "aarch64r_be_exn_rtti",
-            "json": "aarch64r_be_exn_rtti.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armv8-r"
-        },
-        {
-            "variant": "aarch64r_be",
-            "json": "aarch64r_be.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armv8-r -fno-exceptions -fno-rtti"
-        },
-        {
-            "variant": "aarch64r_exn_rtti",
-            "json": "aarch64r_exn_rtti.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -mno-unaligned-access"
-        },
-        {
-            "variant": "aarch64r",
-            "json": "aarch64r.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -fno-exceptions -fno-rtti -mno-unaligned-access"
-        },
-        {
-            "variant": "aarch64r_soft_nofp_exn_rtti_unaligned",
-            "json": "aarch64r_soft_nofp_exn_rtti_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_soft_nofp_unaligned",
-            "json": "aarch64r_soft_nofp_unaligned.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -fno-exceptions -fno-rtti -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_soft_nofp_exn_rtti",
-            "json": "aarch64r_soft_nofp_exn_rtti.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft -mno-unaligned-access",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_soft_nofp",
-            "json": "aarch64r_soft_nofp.json",
-            "flags": "--target=aarch64-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -fno-exceptions -fno-rtti -mabi=aapcs-soft -mno-unaligned-access",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_be_soft_nofp_exn_rtti",
-            "json": "aarch64r_be_soft_nofp_exn_rtti.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
-        {
-            "variant": "aarch64r_be_soft_nofp",
-            "json": "aarch64r_be_soft_nofp.json",
-            "flags": "--target=aarch64_be-unknown-none-elf -march=armv8-r -march=armvX+nofp -march=armvX+nosimd -fno-exceptions -fno-rtti -mabi=aapcs-soft",
-            "libraries_supported": "picolibc"
-        },
         {
             "variant": "armv4t_exn_rtti",
             "json": "armv4t_exn_rtti.json",
diff --git a/arm-software/embedded/arm-multilib/multilib-generate.py b/arm-software/embedded/arm-multilib/multilib-generate.py
index c7ea84c6e093..c57201ee4996 100755
--- a/arm-software/embedded/arm-multilib/multilib-generate.py
+++ b/arm-software/embedded/arm-multilib/multilib-generate.py
@@ -236,13 +236,8 @@ def get_extension_list(clang, triple):
 
 
 def generate_extensions(args):
-    aarch64_features = get_extension_list(args.clang, "aarch64-none-eabi")
     aarch32_features = get_extension_list(args.clang, "arm-none-eabi")
-    all_features = list(aarch64_features)
-    # Combine the aarch64 and aarch32 lists without duplication.
-    # Casting to sets and merging would be simpler, but creates
-    # non-deterministic output.
-    all_features.extend(feat for feat in list(aarch32_features) if feat not in all_features)
+    all_features = list(aarch32_features)
 
     print("# Expand -march=...+[no]feature... into individual options we can match")
     print("# on. We use 'armvX' to represent a feature applied to any architecture, so")
-- 
2.39.5 (Apple Git-154)

